apis/python/tests/test_experiment_query.py

```diff
@@ -43,22 +33,29 @@ def soma_experiment(
     tmp_path,
     n_obs,
     n_vars,
-    obs,
-    var,
     X_layer_names,
     obsp_layer_names,
     varp_layer_names,
 ):
-    return make_experiment(
-        tmp_path,
-        n_obs,
-        n_vars,
-        obs,
-        var,
-        X_layer_names,
-        obsp_layer_names,
-        varp_layer_names,
-    )
+    with soma.Experiment.create((tmp_path / "exp").as_posix()) as exp:
+        add_dataframe(exp, "obs", n_obs)
+        ms = exp.add_new_collection("ms")
+        rna = ms.add_new_collection("RNA", soma.Measurement)
+        add_dataframe(rna, "var", n_vars)
```

Comment:
I conceptually don't understand why this is not

```
rna.add_dataframe("var", n_vars)
```

It seems to me that the `add_dataframe` et al. which you're creating -- as helper functions solely for unit-test -- are the kinds of helper functions _people would actually want to use_. I don't see why we're hiding this good stuff inside unit-test files instead of delivering it and letting people benefit from it.

More generally I think there's an impedance mismatch  between tools created _by_ software engineers, and tools created _for_ computational biologists. The former might be inclined to say _I'm here to give you flexibility! You can separate make-schema from write-data!_ And the latter might be inclined to say _OK I'll use that some fraction of the time but most of the time I want to do both -- why force me to have to write two separate create-schema and write-data function calls in the happy case which form the bulk of my work?_

cc @mojaveazure @ckrilow @aaronwolen -- please let me know if I'm off-base. I'm trying to advocate for users of these tools, not just creators of these tools -- but I'm also in fact really just another software engineer, not a computational biologist. I may be misprepresenting the needs of our intended users.